test/fuzz/gen.ml

```diff
@@ -214,58 +263,88 @@ let rec expr ~block_type ~stack ~locals env =
       in
       drops @ adds
   else
-    let expr_available_with_current_stack =
-      (* TODO: complete this *)
-      match stack with
-      | Num_type I32 :: Num_type I32 :: Num_type I32 :: _tl ->
-        expr_available_1_any
-        @ expr_available_1_i32 if_else expr ~stack ~locals env
-        @ expr_available_2_i32 env @ expr_available_3_i32 env
-      | Num_type I32 :: Num_type I32 :: _tl ->
-        expr_available_1_any
-        @ expr_available_1_i32 if_else expr ~stack ~locals env
-        @ expr_available_2_i32 env
-      | Num_type I32 :: Ref_type (_, Func_ht) :: Num_type I32 :: _tl ->
-        B.table_fill env
-      | Num_type I32 :: Ref_type (_, Func_ht) :: _tl -> B.table_grow env
-      | Ref_type (_, Func_ht) :: Num_type I32 :: _tl -> B.table_set env
-      | Num_type I64 :: Num_type I32 :: _tl -> expr_available_2_i64_i32 env
-      | Num_type F32 :: Num_type I32 :: _tl -> expr_available_2_f32_i32 env
-      | Num_type F64 :: Num_type I32 :: _tl -> expr_available_2_f64_i32 env
-      | Num_type I64 :: Num_type I64 :: _tl ->
-        expr_available_1_any @ expr_available_1_i64 env @ expr_available_2_i64
-      | Num_type I32 :: _tl ->
-        expr_available_1_any
-        @ expr_available_1_i32 if_else expr ~stack ~locals env
-      | Num_type I64 :: _tl -> expr_available_1_any @ expr_available_1_i64 env
-      | Num_type F32 :: Num_type F32 :: _tl ->
-        expr_available_1_any @ expr_available_1_f32 env @ expr_available_2_f32
-      | Num_type F64 :: Num_type F64 :: _tl ->
-        expr_available_1_any @ expr_available_1_f64 env @ expr_available_2_f64
-      | Num_type F32 :: _tl -> expr_available_1_any @ expr_available_1_f32 env
-      | Num_type F64 :: _tl -> expr_available_1_any @ expr_available_1_f64 env
-      | _ -> []
-    in
-    let expr_available env =
-      expr_always_available block loop expr ~locals ~stack env
-      @ expr_available_with_current_stack @ B.expr_call env stack
-      (* TODO: Function calls can be improved: recursive calls are not processed *)
-      @ B.expr_br env stack
-    in
-    let* i, ops = choose (expr_available env) in
-    let stack = S.apply_stack_ops stack ops in
-    let next = expr ~block_type ~stack ~locals env in
-    let i = const i in
-    map [ i; next ] List.cons
+    let* expr_available_1_any in
+    let* expr_available_2_i32 = expr_available_2_i32 env in
+    let* expr_available_2_i64_i32 = expr_available_2_i64_i32 env in
+    let* expr_available_2_f32_i32 = expr_available_2_f32_i32 env in
+    let* expr_available_2_f64_i32 = expr_available_2_f64_i32 env in
+    let* expr_available_3_i32 = expr_available_3_i32 env in
+    let* expr_available_1_i64 = expr_available_1_i64 env in
+    let* expr_available_2_i64 in
+    let* expr_available_1_f32 = expr_available_1_f32 env in
+    let* expr_available_2_f32 in
+    let* expr_available_1_f64 = expr_available_1_f64 env in
+    let* expr_available_2_f64 in
```

Comment:
Why are you using `let* e in` for some and `let* e = e in` for others ? Could you switch all of them with `let* e in` ?

Reply:
Oh indeed, sorry